xdsl/transforms/function_specialization.py

```diff
@@ -0,0 +1,227 @@
+from collections.abc import Iterable
```

Comment:
Can you please add a docstring here, explaining what the pass does, who should use it, and when? I think I have a rough idea but it would be nice to have a little guide. Does MLIR have an equivalent? Or LLVM? Maybe could link to those also?

Reply:
Added docstring to ModulePass!